### PR TITLE
Add support of negative slices

### PR DESCRIPTION
**Pull request by Arman-Ghazaryan**

### CODE REVIEW

File: stringzilla/pybind11.cpp [hunk 1]
```diff
@@ -18,6 +18,7 @@ typedef SSIZE_T ssize_t;
 
 #include <utility> // `std::exchange`
 #include <limits>  // `std::numeric_limits`
+#include <cmath>
```
Comment (ashvardanian) [inline stringzilla/pybind11.cpp:21]: What are you using from here?

File: stringzilla/pybind11.cpp [hunk 2]
```diff
@@ -110,11 +111,73 @@ struct index_span_t {
 };
 
 index_span_t unsigned_slice(size_t length, ssize_t start, ssize_t end) {
-    if (start < 0 || end < 0) // TODO:
-        throw std::invalid_argument("Negative slices aren't supported yet!");
-
-    start = std::min(start, static_cast<ssize_t>(length));
-    end = std::min(end, static_cast<ssize_t>(length));
+    ssize_t len = static_cast<ssize_t>(length);
+    if (start > end)
```
Comment (ashvardanian) [inline stringzilla/pybind11.cpp:115]: Why can't the user write`str[10:-2]`?

**Review by ashvardanian — CHANGES_REQUESTED** [review]

**Comment by ashvardanian** [comment]
Also, please add tests to `test.py`

File: stringzilla/pybind11.cpp (hunk 1 above)
Comment (Arman-Ghazaryan) [reply stringzilla/pybind11.cpp:21]: std::abs, it will be mentioned in the comment at the next commit.

File: stringzilla/pybind11.cpp (hunk 2 above)
Comment (Arman-Ghazaryan) [reply stringzilla/pybind11.cpp]: Fixed.

File: stringzilla/pybind11.cpp [hunk 3]
```diff
@@ -16,8 +16,11 @@ typedef SSIZE_T ssize_t;
 #include <unistd.h> // `ssize_t`
 #endif
 
+#include <iostream>
```
Comment (ashvardanian) [inline stringzilla/pybind11.cpp:19]: Why would you need this? It makes builds much slower.

Comment (Arman-Ghazaryan) [reply stringzilla/pybind11.cpp:19]: Sorry, my fault, it's not supposed to be here.

Comment (Arman-Ghazaryan) [reply stringzilla/pybind11.cpp]: Fixed.

**Pull request merged**